toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -53,24 +67,39 @@ class GoogleBigQueryConnector(ToucanConnector):
         '<a href="https://developers.google.com/identity/protocols/googlescopes" target="_blank" >documentation</a>',
     )
 
-    def _retrieve_data(self, data_source: GoogleBigQueryDataSource) -> pd.DataFrame:
-        """
-        Uses Pandas read_gbq method to extract data from Big Query into a dataframe
-        See: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.io.gbq.read_gbq.html
-        Note:
-            The parameter reauth is set to True to force Google BigQuery to reauthenticate the user
-            for each query. This is necessary when extracting multiple data to avoid the error:
-            [Errno 54] Connection reset by peer
-        """
+    @staticmethod
+    def _get_google_credentials(credentials, scopes):
+        credentials = get_google_oauth2_credentials(credentials).with_scopes(scopes)
+        return credentials
+
+    @staticmethod
+    def _connect(credentials) -> bigquery.Client:
+        start = timer()
+        client = bigquery.Client(credentials=credentials)
+        end = timer()
+        logging.getLogger(__name__).info(
+            f'[benchmark][google_big_query] - connect {end - start} seconds',
+            extra={
+                'benchmark': {
+                    'operation': 'connect',
+                    'execution_time': end - start,
+                    'connector': 'google_big_query',
+                }
+            },
+        )
+        return client
+
+    @staticmethod
+    def _execute_query(client: bigquery.Client, query: str, parameters: List) -> pandas.DataFrame:
         start = timer()
-        data_source.query = apply_query_parameters(data_source.query, data_source.parameters)
-        logging.getLogger(__name__).debug(f'Play request {data_source.query}')
-        credentials = get_google_oauth2_credentials(self.credentials).with_scopes(self.scopes)
-        result = pandas_gbq.read_gbq(
-            query=data_source.query,
-            project_id=self.credentials.project_id,
-            credentials=credentials,
-            dialect=self.dialect,
+        result = (
+            client.query(
+                query, job_config=bigquery.QueryJobConfig(query_parameters=parameters)
+            ).result()
+            # Use to generate directly a dataframe pandas
+            .to_dataframe(
+                create_bqstorage_client=True,
+            )
```

Comment:
Would have suggest : 
```python
).result()
.to_dataframe(
    create_bqstorage_client=True,
) # Use to generate directly a dataframe pandas
```
just to prevent putting comments between object's function calls...